docs/install.md

```diff
@@ -106,27 +126,38 @@ Once the module is installed and enabled, you can check its requirements on
 ![MongoDB on status page](images/mongodb-requirements.png)
 
 
-##Composer Requirements
+###Composer Requirements
 
-* Commands below are for those who are using composer already in your site to
-  manage module dependencies. To know more about composer [here][composer].
+* This section is applicable if you are using composer already in your site to
+  manage module dependencies. Know more about composer [here][composer].
 
 [composer]: https://www.drupal.org/docs/develop/using-composer/using-composer-to-manage-drupal-site-dependencies
 
-* At the root of your site
-  * If you are using the `drupal-composer/drupal-project` skeleton, just add
-    this package:
+  * At the root of your site
+    * If you are using the `drupal-composer/drupal-project` skeleton, just add
+      this package:
+
+         `composer require drupal/mongodb "^2.0.0"`
 
-        composer require drupal/mongodb "^2.0.0"
-  * Otherwise also add a composer requirement by typing:
+    * Otherwise also add a composer requirement by typing:
 
-        composer require drupal/mongodb "^2.0.0"
-        composer require mongodb/mongodb "^1.2.0"
-* If this is the first Composer dependency on the project, from the site root,
-  run:
+        `composer require drupal/mongodb "^2.0.0"`
+        `composer require mongodb/mongodb "^1.2.0"`
 
-        composer install
-* Enable the `mongodb` module. You now have access to the MongoDB services and
-  Drush commands.
+    * If this is the first Composer dependency on the project, run:
+
+        `composer install`
+
+  * Enable the `mongodb` module. You now have access to the MongoDB services and
+  Drush commands for the `mongodb` module.
 
 [install]: https://www.drupal.org/documentation/install/modules-themes/modules-8
+
+Note that there is currently a bug with Composer-based deployment from packages.drupal.org/8 : [#2985860: packages.drupal.org/8 serves incorrect composer.json for module mongodb][composer issue]. The workaround is:
+
+* Either require mongodb/mongodb directly in the root composer.json of the project
+* Add a repositories entry with `"type": "vcs"` and `"url": "https://github.com/fgm/mongodb.git"` in the root `composer.json`, and require the `"drupal/mongodb":"dev-8.x-2.x"` instead of the d.o.-rewritten format for semantic versioning `"drupal/mongodb":"^2.0"`
```

Comment:
While this is correct, it would probably be easier for the reader if is was just a cut-and-paste JSON fragment to add to their composer.json, starting with the "repositories" key.

Reply:
Is this needed still as I can see https://www.drupal.org/project/project_composer/issues/2985860 is closed

Reply:
Yes, it is closed but not resolved: unless you succeed at installing the module with the standard composer method, these steps are needed. Issue 29875860 makes the case for changing the project layout, and this isn't done at the moment, so docs should reflect the actual situation, especially as things won't get better from the d.o. side. as I had hoped they would.